clang/lib/Basic/SourceManager.cpp

```diff
@@ -864,74 +864,7 @@ FileID SourceManager::getFileIDLocal(SourceLocation::UIntTy SLocOffset) const {
 /// This function knows that the SourceLocation is in a loaded buffer, not a
 /// local one.
 FileID SourceManager::getFileIDLoaded(SourceLocation::UIntTy SLocOffset) const {
-  if (SLocOffset < CurrentLoadedOffset) {
-    assert(0 && "Invalid SLocOffset or bad function choice");
```

Comment:
Where would this case end up failing in the new implementation?

Reply:
It would probably fail to find the offset in `ASTReader::GlobalSLocOffsetMap` and fail with more generic error. I'll add this back.